portal/account/index_reset.php

```diff
@@ -0,0 +1,170 @@
+                    </tr>
+                    <tr>
+                        <td><strong><?php echo xlt('Current Password'); ?><strong></td>
+                        <td>
+                            <input class="form-control" name="pass_current" id="pass_current" type="password" required
+                                placeholder="<?php echo xla('Current password to authorize changes.'); ?>"
+                                title="<?php echo xla('Enter your existing current password used to login.'); ?>"
+                                pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}" />
+                        </td>
+                    </tr>
+                    <tr>
+                        <td><strong><?php echo xlt('New or Current Password'); ?><strong></td>
+                        <td>
+                            <input class="form-control" name="pass_new" id="pass_new" type="password" required
+                                placeholder="<?php echo xla('Min length is 8 with upper,lowercase,numbers mix'); ?>"
+                                title="<?php echo xla('You must enter a new or reenter current password to keep it. Even for Username change.'); ?>"
+                                pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}" />
+                        </td>
+                    </tr>
+                    <tr>
+                        <td><strong><?php echo xlt('Confirm Password'); ?><strong></td>
+                        <td>
+                            <input class="form-control" name="pass_new_confirm" id="pass_new_confirm" type="password"
+                                pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}" autocomplete="none" />
+                        </td>
+                    </tr>
+                    <tr>
+                        <td colspan="2"><br><input class="btn btn-primary pull-right" type="submit" name="submit" value="<?php echo xla('Save'); ?>" /></td>
+                    </tr>
+                </table>
+                <div><strong><?php echo '* ' . xlt("All credential fields are case sensitive!") ?></strong></div>
+            </form>
+        <?php } elseif (isset($_POST['submit'])) {
+            if ($auth === false) {
+                unset($_POST['submit']);
+                header("Location: " . $_SERVER['PHP_SELF']);
+            }
+            $code_current = trim($_POST['pass_current']);
+            $plain_code = trim($_POST['pass_new']);
+            $new_salt = oemr_password_salt();
+            $new_hash = oemr_password_hash($plain_code, $new_salt);
+            $sqlUpdatePwd = " UPDATE " . TBL_PAT_ACC_ON . " SET " . COL_POR_PWD . "=?, " . COL_POR_SALT . "=?, " . COL_POR_LOGINUSER . "=?" . " WHERE " . COL_ID . "=?";
+            privStatement($sqlUpdatePwd, array(
+                $new_hash,
+                $new_salt,
+                $_POST['login_uname'],
+                $auth[COL_ID]
+            ));
+            echo "<script>dlgclose();</script>\n";
+        } ?>
```

Comment:
Place the else above, with a error message. Could then place the isset($_POST['submit']) code block at top of script and show error and allow user to still try to complete the form again.